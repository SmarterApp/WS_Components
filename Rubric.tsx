import * as React from 'react';

export interface SampleResponse {
    purpose: string;
    scorePoint: string;
    name: string;
    sampleContent: string;
}

export interface RubricSample {
    maxValue: string;
    minValue: string;
    sampleResponses: SampleResponse[];
}

export interface RubricEntry {
    scorepoint: string;
    name: string;
    value: string;
}

export interface Rubric {
    language: string;
    rubricEntries: RubricEntry[];
    samples: RubricSample[];
}

export class RubricEntryComponent extends React.Component<RubricEntry, {}> {
    render() {
        const pointLabel = this.props.scorepoint === "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorepoint} ${pointLabel})`;
        return (
            <div dangerouslySetInnerHTML={{ __html: this.props.value }} />
        );
    }
}

export class SampleResponseComponent extends React.Component<SampleResponse, {}> {
    render() {
        const pointLabel = this.props.scorePoint == "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorePoint} ${pointLabel})`;
        return (
            <div className="sample-response">
                <b>Purpose: </b> {this.props.purpose} <br />
                <b>Sample Response: </b> <br />
                <div dangerouslySetInnerHTML={{ __html: this.props.sampleContent }} />
            </div>
        );
    }
}

export class RubricComponent extends React.Component<Rubric, {}> {

    renderRubrics() {
        const rubricEntries = this.props.rubricEntries.map((re, i) => <RubricEntryComponent {...re} key={String(i)} />);
        if (rubricEntries.length === 0) {
            return null;
        }

        let rubrics = (
            <div>
                <h4>Rubrics</h4>
                {rubricEntries}
            </div>
        );
        return rubrics;
    }

    renderSampleResponses() {
        let rubricSamples: JSX.Element[] = [];
        let i: number = 0;
        for (const sample of this.props.samples) {
            const key = `${i}:`;
            const responses = sample.sampleResponses.map((sr, idx) => <SampleResponseComponent {...sr} key={key + String(idx)} />);
            rubricSamples.push(...responses);
            i++;
        }
        if (rubricSamples.length === 0) {
            return null;
        }

        let sampleResponses = (
            <div>
                <h4>Sample Responses</h4>
                {rubricSamples}
            </div>
        );

        return sampleResponses;
    }

    render() {
        const label = `${this.props.language} Rubric`;
        return (
            <div className="item-information-content">
                {this.renderRubrics()}
                {this.renderSampleResponses()}
            </div>
        );
    }
}