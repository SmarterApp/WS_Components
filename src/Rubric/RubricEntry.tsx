import * as React from 'react';
import * as AboutItemModels from '../AboutItem/AboutItemModels';
import { Collapsible } from './Collapsible';

export class RubricEntry extends React.Component<AboutItemModels.RubricEntry, {}> {
    render() {
        const pointLabel = this.props.scorepoint === "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorepoint} ${pointLabel})`;
        return (
            <Collapsible label={label}>
                <div dangerouslySetInnerHTML={{ __html: this.props.value }} />
            </Collapsible>
        );
    }
}