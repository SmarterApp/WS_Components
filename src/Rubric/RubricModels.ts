export interface SampleResponseModel {
  purpose: string;
  scorePoint: string;
  name: string;
  sampleContent: string;
}

export interface RubricSampleModel {
  maxValue: string;
  minValue: string;
  sampleResponses: SampleResponseModel[];
}

export interface RubricEntryModel {
  scorepoint: string;
  name: string;
  value: string;
}

export interface RubricModel {
  language: string;
  rubricEntries: RubricEntryModel[];
  samples: RubricSampleModel[];
}
