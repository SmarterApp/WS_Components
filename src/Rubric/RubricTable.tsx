import * as React from "react";
import { RubricModel } from "./RubricModels";

export interface RubricTableProps {
  rubrics: RubricModel[];
}

export class RubricTable extends React.Component<RubricTableProps, {}> {
  renderRubric(rubric: RubricModel, index: number) {
    const rows = rubric.rubricEntries.map(entry => {
      const sample = rubric.samples.find(
        s =>
          s.sampleResponses &&
          s.sampleResponses[0] &&
          s.sampleResponses[0].scorePoint === entry.scorepoint
      );
      const sampleHtml = sample
        ? sample.sampleResponses.map(sr => sr.sampleContent).join("<br/>")
        : "";

      return {
        score: entry.scorepoint,
        rationale: entry.value,
        sample: sampleHtml
      };
    });

    const showSample = rubric.samples.length !== 0;

    const leftAlign = {
      textAlign: "left"
    };

    // tslint:disable:react-no-dangerous-html
    const rowsJsx = rows.map((row, i) => (
      <tr key={i}>
        <td dangerouslySetInnerHTML={{ __html: row.rationale }} />
        <td dangerouslySetInnerHTML={{ __html: row.score }} style={leftAlign} />
        {showSample ? (
          <td
            dangerouslySetInnerHTML={{ __html: row.sample }}
            style={leftAlign}
          />
        ) : (
          undefined
        )}
      </tr>
    ));
    // tslint:enable:react-no-dangerous-html

    return (
      <table className="item-data-table" key={index}>
        <thead>
          <tr>
            <th>Score</th>
            <th>Rationale</th>
            {showSample ? <th>Exemplar</th> : undefined}
          </tr>
        </thead>
        <tbody>{rowsJsx}</tbody>
      </table>
    );
  }

  render() {
    return (
      <div>{this.props.rubrics.map((r, i) => this.renderRubric(r, i))}</div>
    );
  }
}
