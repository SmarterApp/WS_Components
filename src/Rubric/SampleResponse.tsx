import * as React from "react";
import { Collapsible } from "./Collapsible";
import { SampleResponseModel } from "./RubricModels";

export class SampleResponse extends React.Component<SampleResponseModel, {}> {
  render() {
    const pointLabel = this.props.scorePoint == "1" ? "point" : "points";
    const label = `${this.props.name} (${this.props.scorePoint} ${pointLabel})`;
    return (
      <Collapsible label={label}>
        <div className="sample-response">
          <b>Purpose: </b> {this.props.purpose} <br />
          <b>Sample Response: </b> <br />
          <div dangerouslySetInnerHTML={{ __html: this.props.sampleContent }} />
        </div>
      </Collapsible>
    );
  }
}
