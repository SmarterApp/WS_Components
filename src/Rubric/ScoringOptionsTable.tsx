import * as React from "react";
import { SmarterAppOptionModel } from "./RubricModels";

export interface ScoringOptionsProps {
  options: SmarterAppOptionModel[];
}

// tslint:disable-next-line:variable-name
export const ScoringOptions: React.SFC<ScoringOptionsProps> = props => {
  const rowsJSX = props.options.map(o => (
    <tr>
      <td>{o.name}</td>
      <td>{o.feedback}</td>
    </tr>
  ));

  return (
    <table className="item-data-table">
      <thead>
        <tr>
          <th>Option</th>
          <th>Feedback</th>
        </tr>
      </thead>
      <tbody>{rowsJSX}</tbody>
    </table>
  );
};
