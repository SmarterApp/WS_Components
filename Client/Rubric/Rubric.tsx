import * as React from 'react'
import * as AboutItemModels from '../AboutItem/AboutItemModels';
import * as RubricEntry from './RubricEntry';
import * as Collapsible from './Collapsible';
import * as SampleResponse from '../SampleResponse/SampleResponse';

export class RubricComponent extends React.Component<AboutItemModels.Rubric, {}> {

    renderRubrics() {
        const rubricEntries = this.props.rubricEntries.map((re, i) => <RubricEntry.RubricEntryComponent {...re} key={String(i)} />);
        if (rubricEntries.length === 0) {
            return null;
        }

        let rubrics = (
            <div>
                <h4>Rubrics</h4>
                {rubricEntries}
            </div>
        );
        return rubrics;
    }

    renderSampleResponses() {
        let rubricSamples: JSX.Element[] = [];
        let i: number = 0;
    
        for (const sample of this.props.samples) {
            const key = `${i}:`;
            const responses = sample.sampleResponses.map((sr, idx) => <SampleResponse.SampleResponseComponent {...sr} key={key + String(idx)} />);
            rubricSamples.push(...responses);
            i++;
        }

        if (rubricSamples.length === 0) {
            return null;
        }

        let sampleResponses = (
            <div>
                <h4>Sample Responses</h4>
                {rubricSamples}
            </div>
        );

        return sampleResponses;
    }

    render() {
        const label = `${this.props.language} Rubric`;
        return (
            <Collapsible.CComponent label={label}>
                {this.renderRubrics()}
                {this.renderSampleResponses()}
            </Collapsible.CComponent>
        );
    }
}