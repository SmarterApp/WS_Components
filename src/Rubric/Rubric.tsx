import * as React from "react";
import { Collapsible } from "./Collapsible";
import { RubricEntry } from "./RubricEntry";
import { RubricModel } from "./RubricModels";
import { SampleResponse } from "./SampleResponse";

export class Rubric extends React.Component<RubricModel, {}> {
  renderRubrics() {
    const rubricEntries = this.props.rubricEntries.map((re, i) => (
      <RubricEntry {...re} key={String(i)} />
    ));
    if (rubricEntries.length === 0) {
      return null;
    }

    const rubrics = (
      <div>
        <h4>Rubrics</h4>
        {rubricEntries}
      </div>
    );
    return rubrics;
  }

  renderSampleResponses() {
    const rubricSamples: JSX.Element[] = [];
    let i: number = 0;

    for (const sample of this.props.samples) {
      const key = `${i}:`;
      const responses = sample.sampleResponses.map((sr, idx) => (
        <SampleResponse {...sr} key={key + String(idx)} />
      ));
      rubricSamples.push(...responses);
      i++;
    }

    if (rubricSamples.length === 0) {
      return null;
    }

    const sampleResponses = (
      <div>
        <h4>Sample Responses</h4>
        {rubricSamples}
      </div>
    );

    return sampleResponses;
  }

  render() {
    const label = `${this.props.language} Rubric`;
    return (
      <Collapsible label={label}>
        {this.renderRubrics()}
        {this.renderSampleResponses()}
      </Collapsible>
    );
  }
}
