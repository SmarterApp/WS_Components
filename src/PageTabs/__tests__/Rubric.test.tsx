import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as Rubric from '../Rubric';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });



const entry: Rubric.RubricEntry = {
    scorepoint: "123",
    name:  "Steve Harvey",
    value: "val"
}

const sampleResponse: Rubric.SampleResponse = {
    purpose: "purpose",
    scorePoint: "scorePoint",
    name: "name",
    sampleContent: "sampleContent"
}

const samples: Rubric.RubricSample[] =  [];

const rubric: Rubric.Rubric = {
    language: "english",
    rubricEntries: [entry],
    samples
}

describe("RubricEntryComponent", () => {
    it("matches snapshot", () => {
        expect(shallow(<Rubric.RubricEntryComponent {...entry}/>)).toMatchSnapshot();;
    })
})


describe("SampleResponseComponent", () => {
    it("matches snapshot", () => {
        expect(shallow(<Rubric.SampleResponseComponent {...sampleResponse}/>)).toMatchSnapshot();;
    })
})

describe("rubric component", () => {
    it("matches snapshot", () => {
        expect(shallow(<Rubric.RubricComponent {...rubric}/>)).toMatchSnapshot();;
    })
})