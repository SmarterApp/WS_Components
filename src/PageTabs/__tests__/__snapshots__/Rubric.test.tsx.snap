// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RubricEntryComponent matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RubricEntryComponent
    name="Steve Harvey"
    scorepoint="123"
    value="val"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "dangerouslySetInnerHTML": Object {
        "__html": "val",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "dangerouslySetInnerHTML": Object {
          "__html": "val",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SampleResponseComponent matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SampleResponseComponent
    name="name"
    purpose="purpose"
    sampleContent="sampleContent"
    scorePoint="scorePoint"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <b>
          Purpose: 
        </b>,
        " ",
        "purpose",
        " ",
        <br />,
        <b>
          Sample Response: 
        </b>,
        " ",
        <br />,
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "sampleContent",
            }
          }
        />,
      ],
      "className": "sample-response",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Purpose: ",
        },
        "ref": null,
        "rendered": "Purpose: ",
        "type": "b",
      },
      " ",
      "purpose",
      " ",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Sample Response: ",
        },
        "ref": null,
        "rendered": "Sample Response: ",
        "type": "b",
      },
      " ",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "sampleContent",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <b>
            Purpose: 
          </b>,
          " ",
          "purpose",
          " ",
          <br />,
          <b>
            Sample Response: 
          </b>,
          " ",
          <br />,
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "sampleContent",
              }
            }
          />,
        ],
        "className": "sample-response",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Purpose: ",
          },
          "ref": null,
          "rendered": "Purpose: ",
          "type": "b",
        },
        " ",
        "purpose",
        " ",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Sample Response: ",
          },
          "ref": null,
          "rendered": "Sample Response: ",
          "type": "b",
        },
        " ",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "sampleContent",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`rubric component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RubricComponent
    language="english"
    rubricEntries={
      Array [
        Object {
          "name": "Steve Harvey",
          "scorepoint": "123",
          "value": "val",
        },
      ]
    }
    samples={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h4>
            Rubrics
          </h4>
          <RubricEntryComponent
            name="Steve Harvey"
            scorepoint="123"
            value="val"
          />
        </div>,
        null,
      ],
      "className": "item-information-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Rubrics
            </h4>,
            Array [
              <RubricEntryComponent
                name="Steve Harvey"
                scorepoint="123"
                value="val"
              />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Rubrics",
            },
            "ref": null,
            "rendered": "Rubrics",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "name": "Steve Harvey",
              "scorepoint": "123",
              "value": "val",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h4>
              Rubrics
            </h4>
            <RubricEntryComponent
              name="Steve Harvey"
              scorepoint="123"
              value="val"
            />
          </div>,
          null,
        ],
        "className": "item-information-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Rubrics
              </h4>,
              Array [
                <RubricEntryComponent
                  name="Steve Harvey"
                  scorepoint="123"
                  value="val"
                />,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Rubrics",
              },
              "ref": null,
              "rendered": "Rubrics",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "name": "Steve Harvey",
                "scorepoint": "123",
                "value": "val",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
